### PR TITLE
Spring 5 hw 39

### PR DESCRIPTION
**Pull request by yurwing**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        movieSessionService.add(movieSessionMapper.getEntity(movieSessionRequestDto));
+    }
+
+    @DeleteMapping
+    public void deleteMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.delete(movieSessionMapper.getEntity(movieSessionRequestDto));
+    }
+
+    @PutMapping
+    public void updateMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Comment (antoniukivan) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: add id into the request path

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    public void addMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionMapper.getEntity(movieSessionRequestDto));
+    }
+
+    @DeleteMapping
+    public void deleteMovieSession(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Comment (antoniukivan) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: let's delete by id only

**Review by antoniukivan — CHANGES_REQUESTED** [review]

**Review by antoniukivan — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private final CinemaHallService cinemaHallService;
+    private final CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java:22]: let's use the same approach everywhere - either use this annotation over the constructor - or not

File: src/main/java/com/dev/cinema/dao/AbstractDao.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public abstract class AbstractDao<T> {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    protected AbstractDao(SessionFactory sessionFactory) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/AbstractDao.java:13]: https://www.baeldung.com/spring-autowired-abstract-class#constructor-injection
please see this article

File: src/main/java/com/dev/cinema/service/MovieService.java [hunk 5]
```diff
@@ -7,4 +7,7 @@
     Movie add(Movie movie);
 
     List<Movie> getAll();
+
+    Movie getById(Long id);
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/MovieService.java]: redundant empty line

File: src/main/java/com/dev/cinema/service/mapper/Mapper.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.dev.cinema.service.mapper;
+
+public interface Mapper<T, S, U> {
+    S getDto(T t);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/Mapper.java]:
```suggestion
    S getDto(T entity);
```

File: src/main/java/com/dev/cinema/service/mapper/Mapper.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+    T getEntity(U u);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/Mapper.java]:
```suggestion
    T getEntity(U requestDto);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**